src/otx/v2/adapters/torch/mmengine/dataset.py

```diff
@@ -0,0 +1,310 @@
+from otx.v2.adapters.torch.mmengine.registry import MMEngineRegistry
+from otx.v2.adapters.torch.modules.dataloaders import ComposedDL
+from otx.v2.api.entities.task_type import TaskType, TrainType
+from otx.v2.api.utils.decorators import add_subset_dataloader
+from otx.v2.api.utils.type_utils import str_to_subset_type
+
+if TYPE_CHECKING:
+    from mmengine.registry import Registry
+    from torch.utils.data import DataLoader as TorchDataLoader
+    from torch.utils.data import Dataset as TorchDataset
+    from torch.utils.data import Sampler
+
+SUBSET_LIST = ["train", "val", "test"]
+
+
+@add_subset_dataloader(SUBSET_LIST)
+class MMXDataset(TorchBaseDataset):
+    """A class representing a dataset for pretraining a model."""
+
+    def __init__(
+        self,
+        task: TaskType | str | None = None,
+        train_type: TrainType | str | None = None,
+        train_data_roots: str | None = None,
+        train_ann_files: str | None = None,
+        val_data_roots: str | None = None,
+        val_ann_files: str | None = None,
+        test_data_roots: str | None = None,
+        test_ann_files: str | None = None,
+        unlabeled_data_roots: str | None = None,
+        unlabeled_file_list: str | None = None,
+        data_format: str | None = None,
+    ) -> None:
+        r"""MMEngine's Dataset class.
+
+        Args:
+            task (Optional[Union[TaskType, str]], optional): The task type of the dataset want to load.
+                Defaults to None.
+            train_type (Optional[Union[TrainType, str]], optional): The train type of the dataset want to load.
+                Defaults to None.
+            train_data_roots (Optional[str], optional): The root address of the dataset to be used for training.
+                Defaults to None.
+            train_ann_files (Optional[str], optional): Location of the annotation file for the dataset
+                to be used for training. Defaults to None.
+            val_data_roots (Optional[str], optional): The root address of the dataset
+                to be used for validation. Defaults to None.
+            val_ann_files (Optional[str], optional): Location of the annotation file for the dataset
+                to be used for validation. Defaults to None.
+            test_data_roots (Optional[str], optional): The root address of the dataset
+                to be used for testing. Defaults to None.
+            test_ann_files (Optional[str], optional): Location of the annotation file for the dataset
+                to be used for testing. Defaults to None.
+            unlabeled_data_roots (Optional[str], optional): The root address of the unlabeled dataset
+                to be used for training. Defaults to None.
+            unlabeled_file_list (Optional[str], optional): The file where the list of unlabeled images is declared.
+                Defaults to None.
+            data_format (Optional[str], optional): The format of the dataset. Defaults to None.
```

Comment:
An example section here would be useful maybe?

Reply:
I added an Example section. :)